CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/framework/GraphVisualAccess.java

```diff
@@ -1026,24 +1034,68 @@ public ConstellationColor getBlazeColor(final int vertex) {
         return blaze == null ? VisualGraphDefaults.DEFAULT_BLAZE_COLOR : blaze.getColor();
     }
 
+    private String getDecoratorStr(final int decoratorAttrib, final int decoratorVertex) {
+        // If a valid attribute ID has been supplied, return its value as a string. This will be mapped to an icon
+        // for display as a decorator. Boolean attributes are special cases as it is possible to overload the icon
+        // used for true and false values based on attribute name.
+        // This is acheived in the following way:
+        //   * The value string returned for boolean attributes is appended with the text .<attribute_name> to result
+        //     in a value of the form <value>.<attribute_name>
+        //   * In a class overriding ConstellationIcon ensure an icon is added with a name, or an alias matching 
+        //     the string <value>.<attribute_name>
+        // The icon DefaultIconProvider.TRANSPARENT is able to be aliased to support NOT showing an icon. Ie to NOT
+        // show an icon when a pinned attribute is set to false, add an alias to DefaultIconProvider.TRANSPARENT
+        // with name "false.pinned"
+        if (decoratorAttrib != Graph.NOT_FOUND)
+        {
+            final String value = accessGraph.getStringValue(decoratorAttrib, accessGraph.getVertex(decoratorVertex));
+            // If the attribute is a boolean, construct a value string including the attribute name as well,
+            // as per comments above.
+            if (accessGraph.getAttributeType(decoratorAttrib) == BOOLEAN_ATTRIBUTE_STR)
+            {   
+                // Booleans will either have a value of true_<attributeName> or false_<attributeName>
+                // there are three cases to cater for: 
+                // 1. Both true_<attributeName> and false_<attributeName> are set as aliases for icons
+                //    --> In this case we use the supplied icon
+                // 2. Only one of true_<attributeName> or false_<attributeName> is set as an alias for an icon
+                //    --> In this case the value tht doesnt have an icon set should display nothing
+                // 3. Neither true_<attributeName> or false_<attributeName> is set as an alias
+                //    --> In this case there is no override, use the default true/false icons
+                final String notValue = this.bool_dict.get(value);
```

Comment:
@serpens24 can you please consider replacing the 2 value `Map` with a method like this:

```java
private String toggleBooleanValue(final String value){
        if (value == null) {
            return "false";
        } else if ("true".indexOf(value) == 0) {
            return "false";
        } else if ("false".indexOf(value) == 0) {
            return "true";
        } else {
            return "false";
        }
    }
```

I think this is the fastest we can make it and avoiding the need to create a Map to save some memory.

Reply:
Not sure if it's quicker given the boxing/unboxing but could possibly also do something like:
```java
private String toggleBooleanValue(final String value) {
    if (value == null) {
        return "false";
    } else {
        return String.valueOf(!Boolean.valueOf(value));
    }
}
```

Reply:
That looks way neater @antares1470 , thanks for the suggestion. We can go with this as it reads better an profile it sometime if we want.

Reply:
OK , I've done it differently to both suggestions - because we actually have the underlying boolean value so don't need to do these string comparisons.

                final boolean valueAsBool = accessGraph.getBooleanValue(decoratorAttrib, accessGraph.getVertex(decoratorVertex));
                final String notValue = valueAsBool ? "false" : "true";

Reply:
See above. I've taken advantage of the boolean attribute.